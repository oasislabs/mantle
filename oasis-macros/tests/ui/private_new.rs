#![feature(proc_macro_hygiene)]
#[oasis_std::service]
mod service {
    #[derive(Service, Default)]
    pub struct Counter {
        count: u32,
        max_count: u64,
    }

    impl Counter {
        pub(crate) fn new(ctx: &Context) -> Result<Self> {
            Ok(Default::default())
        }
    }
}

fn main() {}
