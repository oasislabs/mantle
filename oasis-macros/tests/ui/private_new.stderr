error: `Counter::new` should have `pub` visibility
  --> $DIR/private_new.rs:10:5
   |
10 |     pub(crate) fn new(ctx: Context) -> Self {
   |     ^^^

error: aborting due to previous error

